java/apache-solr8/Portfile

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                apache-solr8
+version             8.0.0
+categories          java textproc
+maintainers         nomaintainer
+
+description         An open source enterprise search platform.
+
+long_description    Solr is the popular, blazing fast open source enterprise \
+                    search platform from the Apache Lucene project.
+
+homepage            http://lucene.apache.org/solr/
+platforms           darwin
+supported_archs     noarch
+license             Apache-2
+
+dist_subdir         apache-solr
+distname            solr-${version}
+extract.suffix      .tgz
+
+# curl -O https://www.apache.org/dist/lucene/solr/8.0.0/solr-8.0.0.tgz
+# openssl dgst -rmd160 solr-8.0.0.tgz 
+# RIPEMD160(solr-8.0.0.tgz)= 3512cc7298424ea6221ff86b9f0df9bec631d8b2
+# openssl dgst -sha256 solr-8.0.0.tgz 
+# SHA256(solr-8.0.0.tgz)= 0e6392d3b980ab917c731b054101aafcebceacc0e5063cb1e305aeeaec911d12
+
+master_sites        apache:lucene/solr/${version}/
+checksums           rmd160  3512cc7298424ea6221ff86b9f0df9bec631d8b2 \
+                    sha256  0e6392d3b980ab917c731b054101aafcebceacc0e5063cb1e305aeeaec911d12
+
+depends_run         port:openjdk12
+
+set solrGroup       solr
+set solrUser        solr
+add_users           ${solrUser} group=${solrGroup} home=${prefix}/var/solr
+
+# set the destination paths.
+set java_basepath   ${prefix}/share/java
+set solr_destpath   ${java_basepath}/${distname}
+set solr_path       ${solr_destpath}
+# Used to define environment variables for bin/solr
+set solr_home       ${prefix}/var/solr
+set solr_data_home  ${prefix}/var/db/solr
+set solr_logs_dir   ${solr_home}/logs
+set solr_pid_dir    ${prefix}/var/run/solr
+set solr_port       8983
+set solr_heap       1g
+# This is what to set for fine-tuned Java memory control; see bin/solr
+# set solr_java_mem   ""
+set solr_log_prestart_rotation     true
+set solr_ulimit_checks             false
+
+use_configure       no
+
+build {}
+
+destroot {
+    # copy the distribution
+    xinstall -d ${destroot}${java_basepath}
+    copy ${worksrcpath} ${destroot}${java_basepath}
+    # create default directories
+    xinstall -o ${solrUser} -g ${solrGroup} -m 755 -d ${destroot}${prefix}/var/solr
+    xinstall -o ${solrUser} -g ${solrGroup} -m 755 -d ${destroot}${prefix}/var/db/solr
+    xinstall -o ${solrUser} -g ${solrGroup} -m 755 -d ${destroot}${prefix}/var/solr/logs
+    xinstall -o ${solrUser} -g ${solrGroup} -m 775 -d ${destroot}${prefix}/var/run/solr
```

Comment:
You define a number of local variables above, like `set solr_home       ${prefix}/var/solr` why not use them here instead of repeating `${prefix}/var/solr` ?